client/my-sites/upgrades/components/domain-warnings/index.jsx

```diff
@@ -179,15 +205,8 @@ export default React.createClass( {
 		}
 	},
 	render: function() {
-		let pipe = this.getPipe();
-
-		for ( let renderer of pipe ) {
-			let result = renderer();
-			if ( result ) {
-				return result;
-			}
-		}
-		return null;
+		const notices = this.getPipe().map( renderer => renderer() ).filter( notice => notice );
```

Comment:
It was a deliberate decision to show the most important message. If an not-validated email is more important than expiring, you can move that up.

While I agree that it will be good addition to allow multiple message to be visible at any time, it will break the current behaviour and potentially increase the complexity to use.